lib/plugins/unthreaded-signing/plugin.c

```diff
@@ -89,7 +90,7 @@ unthreaded_openssl_sign_hash(sv_unthreaded_plugin_t *self, const uint8_t *hash,
   if (status != SV_OK) return status;
   int idx = self->out_buffer_idx;
 
-  if (status == SV_OK && self->sign_data.signature_size > 0 && idx < MAX_BUFFER_LENGTH) {
+  if (self->sign_data.signature_size > 0 && idx < MAX_BUFFER_LENGTH) {
```

Comment:
I think it is better practice to check availability before signing the hash. There is no point in signing if the generated signature can't be stored.